### ISSUE TITLE
500 error on Ubuntu 18.04 in Docker

### ISSUE THREAD

**Issue by vas3k** [nDmitry/ogimgd#1]
docker-compose build and up were successful, service is running, but when I'm trying to get any preview:

```
GET http://og.vas3k.club/preview?title=test&author=vas3k&bg=https://i.vas3k.club/88481c444687acc4920bc80beac3629c18ec35c1230b5a91b92e1142afd4e5f4.jpg&ava=https://i.vas3k.club/88481c444687acc4920bc80beac3629c18ec35c1230b5a91b92e1142afd4e5f4.jpg&icon=https://i.vas3k.club/88481c444687acc4920bc80beac3629c18ec35c1230b5a91b92e1142afd4e5f4.jpg
```

I get an empty page and this error in console:

```
2021/05/22 19:55:26 getting a resource: https://i.vas3k.club/88481c444687acc4920bc80beac3629c18ec35c1230b5a91b92e1142afd4e5f4.jpg
2021/05/22 19:55:26 getting a resource: https://i.vas3k.club/88481c444687acc4920bc80beac3629c18ec35c1230b5a91b92e1142afd4e5f4.jpg
2021/05/22 19:55:26 getting a resource: https://i.vas3k.club/88481c444687acc4920bc80beac3629c18ec35c1230b5a91b92e1142afd4e5f4.jpg
2021/05/22 19:55:26 Resizing an image to 1200x630 px

 panic: could not resize the background: jpegsave_buffer: no property named `subsample_mode'

Stack:
goroutine 44 [running]:
runtime/debug.Stack(0x7fea98f940a0, 0xc000186b00, 0x34)
	/usr/lib/go/src/runtime/debug/stack.go:24 +0x9f
github.com/davidbyttow/govips/v2/vips.handleVipsError(0x7b2b20, 0xc4bf88)
	/root/go/pkg/mod/github.com/davidbyttow/govips/v2@v2.6.0/vips/error.go:38 +0x58
github.com/davidbyttow/govips/v2/vips.handleSaveBufferError(0x0, 0xc0ffffffff, 0xc0000e2e90)
	/root/go/pkg/mod/github.com/davidbyttow/govips/v2@v2.6.0/vips/error.go:31 +0x2c
github.com/davidbyttow/govips/v2/vips.vipsSaveToBuffer(0x151f330, 0x0, 0x1, 0x0, 0x5000000000, 0x1, 0x0, 0x0, 0x600000000, 0x8, ...)
	/root/go/pkg/mod/github.com/davidbyttow/govips/v2@v2.6.0/vips/foreign.go:347 +0xbe
github.com/davidbyttow/govips/v2/vips.vipsSaveJPEGToBuffer(0x151f330, 0x0, 0x50, 0x1, 0x0, 0x0, 0x0, 0x13, 0xc0007200d0, 0xc0001a6a40, ...)
	/root/go/pkg/mod/github.com/davidbyttow/govips/v2@v2.6.0/vips/foreign.go:283 +0x138
github.com/davidbyttow/govips/v2/vips.(*ImageRef).ExportJpeg(0xc0001a6a40, 0xc0000e31a8, 0x44927c, 0xc0001a8c00, 0x300000002, 0xc0001a8c00, 0x0, 0x0)
	/root/go/pkg/mod/github.com/davidbyttow/govips/v2@v2.6.0/vips/image.go:538 +0xa5
github.com/davidbyttow/govips/v2/vips.(*ImageRef).ExportNative(0xc0001a6a40, 0xc0000e3220, 0x54c39c, 0x780240, 0xc0000e3230, 0xc000000000, 0xc0000e3230)
	/root/go/pkg/mod/github.com/davidbyttow/govips/v2@v2.6.0/vips/image.go:518 +0x20c
github.com/davidbyttow/govips/v2/vips.(*ImageRef).Export(0xc0001a6a40, 0xc0000e3358, 0x276, 0x3, 0x0, 0x0, 0x838926, 0x4)
	/root/go/pkg/mod/github.com/davidbyttow/govips/v2@v2.6.0/vips/image.go:456 +0x58
github.com/nDmitry/ogimgd/internal/preview.resize(0xc000502500, 0x2246, 0x2500, 0x4b0, 0x276, 0x0, 0x0, 0x0, 0x0, 0x0)
	/build/internal/preview/preview.go:303 +0x2b3
github.com/nDmitry/ogimgd/internal/preview.(*Preview).drawBackground(0xc000056520, 0xc000502500, 0x2246, 0x2500, 0x3, 0x3)
	/build/internal/preview/preview.go:127 +0x68
github.com/nDmitry/ogimgd/internal/preview.(*Preview).Draw(0xc000056520, 0xa49448, 0xc0007160c0, 0x4b0, 0x276, 0x3fe3333333333333, 0x40, 0xc0001d8013, 0x4, 0x4053000000000000, ...)
	/build/internal/preview/preview.go:99 +0x27e
github.com/nDmitry/ogimgd/internal/server.getPreview.func1(0x7fea6f1e68b8, 0xc0001a6280, 0xc000129300)
	/build/internal/server/handlers.go:110 +0x455
net/http.HandlerFunc.ServeHTTP(0xc00000e210, 0x7fea6f1e68b8, 0xc0001a6280, 0xc000129300)
	/usr/lib/go/src/net/http/server.go:2069 +0x44
github.com/go-chi/chi.(*Mux).routeHTTP(0xc000062ae0, 0x7fea6f1e68b8, 0xc0001a6280, 0xc000129300)
	/root/go/pkg/mod/github.com/go-chi/chi@v1.5.4/mux.go:436 +0x28b
net/http.HandlerFunc.ServeHTTP(0xc0000211f0, 0x7fea6f1e68b8, 0xc0001a6280, 0xc000129300)
	/usr/lib/go/src/net/http/server.go:2069 +0x44
github.com/go-chi/chi/middleware.Recoverer.func1(0x7fea6f1e68b8, 0xc0001a6280, 0xc000129300)
	/root/go/pkg/mod/github.com/go-chi/chi@v1.5.4/middleware/recoverer.go:37 +0x83
net/http.HandlerFunc.ServeHTTP(0xc00000e228, 0x7fea6f1e68b8, 0xc0001a6280, 0xc000129300)
	/usr/lib/go/src/net/http/server.go:2069 +0x44
github.com/go-chi/chi/middleware.RequestLogger.func1.1(0xa48ca0, 0xc0001460e0, 0xc000129000)
	/root/go/pkg/mod/github.com/go-chi/chi@v1.5.4/middleware/logger.go:57 +0x2e2
net/http.HandlerFunc.ServeHTTP(0xc000081830, 0xa48ca0, 0xc0001460e0, 0xc000129000)
	/usr/lib/go/src/net/http/server.go:2069 +0x44
github.com/go-chi/chi/middleware.RealIP.func1(0xa48ca0, 0xc0001460e0, 0xc000129000)
	/root/go/pkg/mod/github.com/go-chi/chi@v1.5.4/middleware/realip.go:34 +0x9d
net/http.HandlerFunc.ServeHTTP(0xc00000e240, 0xa48ca0, 0xc0001460e0, 0xc000129000)
	/usr/lib/go/src/net/http/server.go:2069 +0x44
github.com/go-chi/chi/middleware.RequestID.func1(0xa48ca0, 0xc0001460e0, 0xc000128d00)
	/root/go/pkg/mod/github.com/go-chi/chi@v1.5.4/middleware/request_id.go:76 +0x1eb
net/http.HandlerFunc.ServeHTTP(0xc00000e258, 0xa48ca0, 0xc0001460e0, 0xc000128d00)
	/usr/lib/go/src/net/http/server.go:2069 +0x44
github.com/go-chi/chi.(*Mux).ServeHTTP(0xc000062ae0, 0xa48ca0, 0xc0001460e0, 0xc000128b00)
	/root/go/pkg/mod/github.com/go-chi/chi@v1.5.4/mux.go:87 +0x331
net/http.serverHandler.ServeHTTP(0xc000146000, 0xa48ca0, 0xc0001460e0, 0xc000128b00)
	/usr/lib/go/src/net/http/server.go:2887 +0xa3
net/http.(*conn).serve(0xc0000ccc80, 0xa49480, 0xc0001a6180)
	/usr/lib/go/src/net/http/server.go:1952 +0x8cd
created by net/http.(*Server).Serve
	/usr/lib/go/src/net/http/server.go:3013 +0x39b


 -> github.com/nDmitry/ogimgd/internal/server.getPreview.func1
 ->   /build/internal/server/handlers.go:113

    net/http.HandlerFunc.ServeHTTP
      /usr/lib/go/src/net/http/server.go:2069
    github.com/go-chi/chi.(*Mux).routeHTTP
      /root/go/pkg/mod/github.com/go-chi/chi@v1.5.4/mux.go:436
    net/http.HandlerFunc.ServeHTTP
      /usr/lib/go/src/net/http/server.go:2069
    github.com/go-chi/chi/middleware.Recoverer.func1
      /root/go/pkg/mod/github.com/go-chi/chi@v1.5.4/middleware/recoverer.go:37
    net/http.HandlerFunc.ServeHTTP
      /usr/lib/go/src/net/http/server.go:2069
    github.com/go-chi/chi/middleware.RequestLogger.func1.1
      /root/go/pkg/mod/github.com/go-chi/chi@v1.5.4/middleware/logger.go:57
    net/http.HandlerFunc.ServeHTTP
      /usr/lib/go/src/net/http/server.go:2069
    github.com/go-chi/chi/middleware.RealIP.func1
      /root/go/pkg/mod/github.com/go-chi/chi@v1.5.4/middleware/realip.go:34
    net/http.HandlerFunc.ServeHTTP
      /usr/lib/go/src/net/http/server.go:2069
    github.com/go-chi/chi/middleware.RequestID.func1
      /root/go/pkg/mod/github.com/go-chi/chi@v1.5.4/middleware/request_id.go:76
    net/http.HandlerFunc.ServeHTTP
      /usr/lib/go/src/net/http/server.go:2069
    github.com/go-chi/chi.(*Mux).ServeHTTP
      /root/go/pkg/mod/github.com/go-chi/chi@v1.5.4/mux.go:87
    net/http.serverHandler.ServeHTTP
      /usr/lib/go/src/net/http/server.go:2887
    net/http.(*conn).serve
      /usr/lib/go/src/net/http/server.go:1952
    created by net/http.(*Server).Serve
      /usr/lib/go/src/net/http/server.go:3013

2021/05/22 19:55:26 [69722edfac52/WeA8HmHDdO-000011] "GET http://og.vas3k.club/preview?title=test&author=vas3k&bg=https://i.vas3k.club/88481c444687acc4920bc80beac3629c18ec35c1230b5a91b92e1142afd4e5f4.jpg&ava=https://i.vas3k.club/88481c444687acc4920bc80beac3629c18ec35c1230b5a91b92e1142afd4e5f4.jpg&icon=https://i.vas3k.club/88481c444687acc4920bc80beac3629c18ec35c1230b5a91b92e1142afd4e5f4.jpg HTTP/1.0" from 95.91.245.225 - 500 0B in 148.645976ms
```

I have no idea why. Trying to investigate it.

Maybe it's something wrong with my Go version. I had troubles building it on the first try.

**Comment by vas3k** [comment]
Ok, looks like there's a problem with my local environment. Hardcoding alpine version helps: https://github.com/LionZXY/ogimgd/blob/main/Dockerfile

**Comment by nDmitry** [comment]
@vas3k I fixed images versions and moved to golang 1.16 for the first stage of building (since `embed` was introduced in 1.16), does it work now?

**Comment by vas3k** [comment]
It works with hardcoded version, thanks!
